test/geometry/calibration/test_undistort.py

```diff
@@ -0,0 +1,181 @@
+import pytest
+import torch
+from torch.autograd import gradcheck
+from torch.testing import assert_allclose
+
+from kornia.geometry.calibration.undistort import undistort_points
+
+
+class TestUndistortion:
+    def test_smoke(self, device, dtype):
+        points = torch.rand(1, 2, device=device, dtype=dtype)
+        K = torch.rand(3, 3, device=device, dtype=dtype)
+        distCoeff = torch.rand(4, device=device, dtype=dtype)
+        pointsu = undistort_points(points, K, distCoeff)
+        assert points.shape == pointsu.shape
+
+        points = torch.rand(1, 1, 2, device=device, dtype=dtype)
+        K = torch.rand(1, 3, 3, device=device, dtype=dtype)
+        distCoeff = torch.rand(1, 4, device=device, dtype=dtype)
+        pointsu = undistort_points(points, K, distCoeff)
+        assert points.shape == pointsu.shape
+
+    @pytest.mark.parametrize(
+        "batch_size, num_points, num_distcoeff", [(1, 3, 4), (2, 4, 5), (3, 5, 8), (4, 6, 12), (5, 7, 14)]
+    )
+    def test_shape(self, batch_size, num_points, num_distcoeff, device, dtype):
+        B, N, Ndist = batch_size, num_points, num_distcoeff
+
+        points = torch.rand(B, N, 2, device=device, dtype=dtype)
+        K = torch.rand(B, 3, 3, device=device, dtype=dtype)
+        distCoeff = torch.rand(B, Ndist, device=device, dtype=dtype)
+
+        pointsu = undistort_points(points, K, distCoeff)
+        assert points.shape == (B, N, 2)
+
+    def test_opencv(self, device, dtype):
+        # ----- Test 1: using 5 distortion coefficients
```

Comment:
Also Split in three different tests
test_opencv_five_coeff
test_opencv_all_coeff
test_opencv_stereo